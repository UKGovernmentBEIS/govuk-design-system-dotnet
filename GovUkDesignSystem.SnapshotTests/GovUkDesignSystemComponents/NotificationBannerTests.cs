using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.GovUkDesignSystemComponents.SubComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class CNotificationBannerTests : SnapshotTestBase
    {
        private NotificationBannerViewModel DefaultNotificationBannerViewModel()
        {
            return new NotificationBannerViewModel
            {
                Text = "Banner text",
                TitleText = "Title text",
                TitleHeadingLevel = "3",
                Type = "success",
                TitleId = "title-id",
                DisableAutoFocus = true,
                Classes = "custom-class"
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("NotificationBanner", DefaultNotificationBannerViewModel());
        }
    }
}