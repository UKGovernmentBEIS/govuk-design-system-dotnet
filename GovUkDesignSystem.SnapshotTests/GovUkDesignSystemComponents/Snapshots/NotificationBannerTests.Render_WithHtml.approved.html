<div class="govuk-notification-banner govuk-notification-banner--success custom-class" role="region" aria-labelledby="title-id" data-module="govuk-notification-banner" data-disable-auto-focus="true">
  <div class="govuk-notification-banner__header">
    <h3 class="govuk-notification-banner__title" id="title-id">

html title        </h3>
  </div>
  <div class="govuk-notification-banner__content">

html text    </div>
</div>